### ISSUE TITLE
Is it expected to lose rows when partitioning with some partition values being `None`?

### ISSUE THREAD

**Issue by yohplala** [dask/fastparquet#711]
**What happened**:
Checking a test case, I see that if some partition values are `None`, corresponding row group is not created. See test `test_groups_roundtrip` in `test_output.py`.

```python
def test_groups_roundtrip(tempdir, scheme):
    df = pd.DataFrame({'a': np.random.choice(['a', 'b', None], size=1000),
                       'b': np.random.randint(0, 64000, size=1000),
                       'c': np.random.choice([True, False], size=1000)})
    writer.write(tempdir, df, partition_on=['a', 'c'], file_scheme=scheme)

    r = ParquetFile(tempdir)

len(r.to_pandas())
Out[44]: 677

lend(df)
Out[45]: 1000
```
I could check in temp directory that if value in column `a` is `None`, row groups are not created.
In my test, end of DataFrame looks like this (before recording).
```python
        a      b      c
997     b  36423   True
998  None  36813  False
999     b  50971   True
```
When loading it back, it looks like.
```python
         b  a      c
674  32475  b   True
675  36423  b   True
676  50971  b   True
```
We see the row with `None` is removed.

So my question is: 'is it expected?'

I am a bit surprised. I would have expected one of the following:
- either a partition value `None` is possible
- or if not, that an error is raised to warn the use that some data will be lost if written, and to use a parameter to force writing of data, no matter what.

**What you expected to happen**:
An error message maybe

**Environment**:

- Fasparquet version: 0.temp.1 (last from source at time of writing the ticket)
- Python version: 
- Operating System: Ubuntu 20.04
- Install method: source

**Comment by martindurant** [comment]
Correct, pandas does not, by default, make a group for None, and I'm not aware if a parquet convention for encoding it in path names.
